### ISSUE TITLE
Sample: transfer data (sensitive api) from background to popup

### ISSUE THREAD

**Issue by hyfydistro** [GoogleChrome/chrome-extensions-samples#774]
Currently, I'm unable to access the Chrome Extension Group so I will post here.

I'm just wondering if anyone knows how to grab data from the service worker (background script) and receive it at the popup page. I am using airtable and it has some sensitive information for the configuration I don't want to leak out so I have it setup in my background script. I just want to transfer certain data and display it on the client safely.

Is there any available examples?

**Comment by guest271314** [comment]
> I just want to transfer certain data and display it on the client safely.

How are you defining "safely"?

You can use messaging to send data to and from `ServiceWorker` from content script.

**Comment by Hanaffi** [comment]
@patrickkettner @AmySteam can you assign it to me please?

**Comment by oliverdunk** [comment]
@Hanaffi, could you share some more about what you are proposing for a sample?

**Comment by Hanaffi** [comment]
@oliverdunk we can do message passing using `chrome.tabs.sendMessage` and listening for it on the other side.
We can also do asymmetric encryption of the message but I think it would be an overkill because I think this message passing API is safe enough, thoughts?

**Comment by Hanaffi** [comment]
@oliverdunk ?

**Comment by oliverdunk** [comment]
Agreed that encryption would be too much here. We already have an [Action API demo](https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/api-samples/action). Perhaps you could update that with some simple functionality where the popup gets some data by messaging the background?
